app/models/user.js

```diff
@@ -266,6 +266,32 @@ exports.fetchByUid = exports.model = function (uid, handler) {
   );
 };
 
+/**
+ * @type {(uid : import('mongodb').ObjectId | string) => Promise<Partial<UserDocument>> }
+ */
+exports.fetchAndProcessUserById = async function (uid) {
+  const user = await mongodb.collections['user'].findOne({
+    _id: typeof uid == 'string' && isObjectId(uid) ? ObjectId(uid) : uid,
+  });
+  if (user) {
+    user.id = '' + user._id;
+    user.mid = '/u/' + user.id;
+    processUsers([user]);
```

Comment:
TODO: it may not be necessary to call `processUsers()` here

Reply:
Actually it must be, because `cacheUsers()` was calling `userModel.fetchMulti()`, which calls `processUsers()`.